hometask-1-OOP/src/Shape.ts

```diff
@@ -1,3 +1,58 @@
+import { Point } from './Point';
+
 export abstract class Shape {
-  abstract getType(): string;
+	protected points: Point[];
+	protected color: string;
+	protected filled: any;
+
+	constructor(points: Point[]);
+	constructor(points: Point[], color: string, filled: boolean);
+
+	constructor(points: Point[], color: string = 'green', filled: boolean = true) {
+		if (points.length < 3) {
+			throw new Error('Please provide at least 3 points to make a shape');
+		}
+
+		this.points = points;
+		this.color = color;
+		this.filled = filled;
+	}
+
+	abstract getType(): string;
+
+	public toString() {
+		const pointsStringified = this.points.map((point: Point): string => point.toString()).join(', ');
```

Comment:
you can move this logic to a private method.

Reply:
yup, thanks. 

Could there be any issues regarding of inheritance of private methods from abstract class.